invenio_jsonschemas/ext.py

```diff
@@ -0,0 +1,167 @@
+            if dir_path == '.':
+                dir_path = ''
+            for file in files:
+                if file.lower().endswith((".json")):
+                    schema_name = os.path.join(dir_path, file)
+                    if schema_name in self.schemas:
+                        raise JSONSchemaDuplicate(
+                            schema_name,
+                            self.schemas[schema_name],
+                            directory
+                        )
+                    self.schemas[schema_name] = os.path.abspath(directory)
+
+    def register_schema(self, directory, path):
+        """Register a json-schema.
+
+        :param directory: root directory path.
+        :param path: schema path, relative to the root directory.
+        """
+        self.schemas[path] = os.path.abspath(directory)
+
+    def get_schema_dir(self, path):
+        """Retrieve the directory containing the given schema.
+
+        :param path: schema path, relative to the directory when it was
+        registered.
+        """
+        if path not in self.schemas:
+            raise JSONSchemaNotFound(path)
+        return self.schemas[path]
+
+    def get_schema_path(self, path):
+        """Compute the schema's absolute path from a schema relative path.
+
+        :param path: relative path of the schema.
+        """
+        if path not in self.schemas:
+            raise JSONSchemaNotFound(path)
+        return os.path.join(self.schemas[path], path)
+
+    def get_schema(self, path):
+        """Retrieve a schema.
+
+        :param path: schema's relative path.
+        """
+        # TODO: add a cache
+        if path not in self.schemas:
+            raise JSONSchemaNotFound(path)
+        with open(os.path.join(self.schemas[path], path)) as file:
+            return json.load(file)
+
+    def list_schemas(self):
+        """List all JSON-schema names.
+
+        :Return: list of schema names.
+        :rtype: list
+        """
+        return self.schemas.keys()
+
+
+class InvenioJSONSchemas(object):
+    """Invenio-JSONSchemas extension.
+
+    Register blueprint serving registered schemas and can be used as an api
+    to register those schemas.
+
+    Note: JSON schemas are served as static files. Thus their "id" and "$ref"
+    fields might not match the Flask application's host and port.
+    """
+
+    CONFIG_ENTRY_POINT_GROUP = 'JSONSCHEMAS_ENTRY_POINT_GROUP'
+    CONFIG_ENDPOINT = 'JSONSCHEMAS_ENDPOINT'
+
+    def __init__(self, app=None):
+        """Extension initialization."""
+        if app:
+            self._state = self.init_app(app)
+
+    def init_app(self, app):
+        """Flask application initialization."""
+        self.init_config(app)
+
+        state = InvenioJSONSchemasState(app)
+
+        entry_point_group = \
+            app.config[InvenioJSONSchemas.CONFIG_ENTRY_POINT_GROUP]
+
+        # Load the json-schemas from extension points.
+        if entry_point_group:
```

Comment:
I think it is best if we follow the same pattern for how we load entry points. Most other extensions today passes the entry point group as an argument to the initialisation and not as config parameters. See e.g. https://github.com/inveniosoftware/invenio-db/blob/master/invenio_db/core.py

Is there some particular problems solved using the config to define the entry point?